src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -443,5 +458,182 @@ internal static bool ContainsPathComponent(string path, string component, bool i
 
             return false;
         }
+
+        /// <summary>
+        /// Gets a path relative to a directory.
+        /// </summary>
+        public static string GetRelativePath(string directory, string fullPath)
```

Comment:
The directory is a path to a directory.  The fullpath is a path to some other file.  The relative path is a path to that file relative to the directory as the origin position.